Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp

```diff
@@ -216,49 +255,67 @@ std::unique_ptr<DisplayVBlankMonitor> DisplayVBlankMonitorDRM::create(PlatformDi
     }
 #endif
 
+    UnixFileDescriptor fd;
 #if PLATFORM(WPE) && ENABLE(WPE_PLATFORM)
-    String filename;
-    if (usingWPEPlatformAPI)
-        filename = String::fromUTF8(wpe_display_get_drm_device(wpe_display_get_primary()));
-    else
-        filename = WebCore::PlatformDisplay::sharedDisplay().drmDeviceFile();
-#else
-    auto filename = WebCore::PlatformDisplay::sharedDisplay().drmDeviceFile();
-#endif
-    if (filename.isEmpty()) {
-        RELEASE_LOG_FAULT(DisplayLink, "Could not create a vblank monitor for display %u: no DRM device found", displayID);
-        return nullptr;
-    }
+#ifdef WPE_PLATFORM_DRM
+    if (usingWPEPlatformAPI && WPE_IS_MONITOR_DRM(monitor)) {
+        String filename = String::fromUTF8(wpe_display_get_drm_device(wpe_display_get_primary()));
 
-    auto fd = UnixFileDescriptor { open(filename.utf8().data(), O_RDWR | O_CLOEXEC), UnixFileDescriptor::Adopt };
-    if (!fd) {
-        RELEASE_LOG_FAULT(DisplayLink, "Could not create a vblank monitor for display %u: failed to open %s", displayID, filename.utf8().data());
-        return nullptr;
+        if (filename.isEmpty()) {
+            RELEASE_LOG_FAULT(DisplayLink, "Could not create a vblank monitor for display %u: no DRM device found", displayID);
+            return nullptr;
+        }
+
+        fd = UnixFileDescriptor(open(filename.utf8().data(), O_RDWR | O_CLOEXEC), UnixFileDescriptor::Adopt);
+        if (!fd) {
+            RELEASE_LOG_FAULT(DisplayLink, "Could not create a vblank monitor for display %u: failed to open %s", displayID, filename.utf8().data());
+            return nullptr;
+        }
```

Comment:
For the case of DRM, I think it would be simpler if we create a drmNodeWithCrtcInfo, initialized with the fd here, and adding later the crtc info, that way we can handle it the same way and avoid more ifdefs.